pkg/lib/tmp/copy.go

```diff
@@ -0,0 +1,54 @@
+package tmp
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// CopyTmpDB reads the file at the given path and copies it to a tmp directory, returning the copied file path or an err
+func CopyTmpDB(original string) (path string, err error) {
+	dst, err := ioutil.TempFile("", "db-")
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if cerr := dst.Close(); cerr != nil && err == nil {
+			err = cerr
+		}
+	}()
+
+	src, err := OpenRegularFile(original)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if cerr := src.Close(); cerr != nil && err == nil {
```

Comment:
IMO it's better for the registry server to tolerate errors from closing `src`, but that would either limit the general usefulness of this package or add some way to make that determination by inspecting the error value. Hopefully it's rare enough to not matter.